### PR TITLE
PIL-462 PIL-534 send amend group details and contact details

### PR DESCRIPTION
**Pull request by javed-iqbal1**

### CODE REVIEW

File: app/connectors/AmendSubscriptionConnector.scala [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, HttpResponse, NotFoundException, UpstreamErrorResponse}
+
+import java.io.IOException
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class AmendSubscriptionConnector @Inject() (val userAnswersConnectors: UserAnswersConnectors, val config: FrontendAppConfig, val http: HttpClient)
+    extends Logging {
+
+  val subscriptionUrl = s"${config.pillar2BaseUrl}/report-pillar2-top-up-taxes/subscription/amend-subscription"
```
Comment (jassalrichy) [inline app/connectors/AmendSubscriptionConnector.scala:33]: Minor, ideally should be named amendSubscriptionUrl

File: app/controllers/subscription/manageAccount/ManageContactCheckYourAnswersController.scala [hunk 2]
```diff
@@ -72,4 +76,15 @@ class ManageContactCheckYourAnswersController @Inject() (
       Redirect(controllers.routes.BookmarkPreventionController.onPageLoad)
     }
   }
+
+  def onSubmit(): Action[AnyContent] = (identify andThen getData andThen requireData) async { implicit request =>
+    amendSubscriptionService.amendSubscription(AmendSubscriptionRequestParameters(request.userId)).flatMap {
+      case Right(s) =>
+        userAnswersConnectors.remove(request.userId)
```
Comment (jassalrichy) [inline app/controllers/subscription/manageAccount/ManageContactCheckYourAnswersController.scala:83]: Question: If two users are updating the same pillar2 record, would this potentially throw an error if one update has been applied before the second user submits there changes (as the record world have already been removed from userAnswers) ?

File: app/views/subscriptionview/manageAccount/ManageGroupDetailsCheckYourAnswersView.scala.html [hunk 3]
```diff
@@ -19,26 +19,27 @@
...
+    govukButton: GovukButton,
+    formHelper: FormWithCSRF,
 )
 
 
 @(list: SummaryList)(implicit request: Request[_], appConfig: FrontendAppConfig, messages: Messages)
 
 @layout(pageTitle = titleNoForm(messages("checkYourAnswers.title")),showBackLink = false) {
+    @formHelper(action = controllers.subscription.manageAccount.routes.ManageGroupDetailsCheckYourAnswersController.onSubmit) {
+        <h1 class="govuk-heading-l">@messages("ManageFurtherGroupDetail.checkYourAnswers.heading")</h1>
```
Comment (jassalrichy) [inline app/views/subscriptionview/manageAccount/ManageGroupDetailsCheckYourAnswersView.scala.html:32]: Key ManageFurtherGroupDetail, this should be in camel case manageGroupDetails within messages file.

File: app/views/subscriptionview/manageAccount/ManageGroupDetailsCheckYourAnswersView.scala.html [hunk 4]
```diff
@@ -19,26 +19,27 @@
...
 
-    <h1 class="govuk-heading-l">@messages("ManageFurtherGroupDetail.checkYourAnswers.heading")</h1>
+        @govukSummaryList(list)
 
-    @govukSummaryList(list)
 
+        <br>
+        @govukButton(
+            ButtonViewModel(messages("site.mange.save-and-continue")).withAttribute("id", "submit")
```
Comment (jassalrichy) [inline app/views/subscriptionview/manageAccount/ManageGroupDetailsCheckYourAnswersView.scala.html:38]: mange, should this be manage ?

File: app/controllers/subscription/manageAccount/ManageContactCheckYourAnswersController.scala (hunk 2 above)
Comment (javed-iqbal1) [reply app/controllers/subscription/manageAccount/ManageContactCheckYourAnswersController.scala:83]: There will be only one user who using its creds to amend.

**Pull request merged**